### ISSUE TITLE
Typo in paper?

### ISSUE THREAD

**Issue by PatWie** [martinarjovsky/WassersteinGAN#5]
I am not sure if I miss something.

Looking at the source gives:
- update D by [`f(real) - f(g(prior samples))`](https://github.com/martinarjovsky/WassersteinGAN/blob/master/main.py#L197)
- update G by [`f(g(prior samples))`](https://github.com/martinarjovsky/WassersteinGAN/blob/master/main.py#L208)

Looking at the paper gives:
- update D by `f(real) - f(g(prior samples))` in Algorithm 1 line 5
- update G by  `- f(g(prior samples))` in Algorithm 1 line 10

Is the minus sign correct in Algorithm 1 line 10?

**Comment by soumith** [comment]
when we update G, we backprop with -1 gradients:
https://github.com/martinarjovsky/WassersteinGAN/blob/master/main.py#L196

**Comment by PatWie** [comment]
Thank you. Sorry for asking.

**Comment by sguada** [comment]
Actually it seems that when you update the Generator is here
https://github.com/martinarjovsky/WassersteinGAN/blob/master/main.py#L212

Which backprop with 1 gradients.

**Comment by PatWie** [comment]
Yeah, I realized that it es simply the pytorch way of implementing
https://github.com/PatWie/tensorpack-fork/blob/wgan/examples/GAN/GAN.py#L96-L100
